src/main/java/duke/operation/TaskList.java

```diff
@@ -0,0 +1,214 @@
+package duke.operation;
+
+import duke.Parser.Parser;
+import duke.exception.DukeException;
+
+import java.util.ArrayList;
+
+/**
+ * This is the TaskList class to contain task list and handle operations.
+ */
+public class TaskList {
```

Comment:
This is a huge class, simultaneously maintaining the list of tasks and handling the I/O too... would it be possible to maybe delegate some of its responsibilities?